tests/k8s/tests/04-bad-cnp-import.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+# This tests:
+# Does the agent crash when it receives a malformed CNP from k8s 
+
+
+dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+source "${dir}/../helpers.bash"
+# dir might have been overwritten by helpers.bash
+dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+
+source "${dir}/../cluster/env.bash"
+
+TEST_NAME=$(get_filename_without_extension $0)
+LOGS_DIR="${dir}/cilium-files/${TEST_NAME}/logs"
+redirect_debug_logs ${LOGS_DIR}
+
+set -ex
+
+bad_policy_path="${dir}/deployments/bad-cnp"
+
+NAMESPACE="kube-system"
+LOCAL_CILIUM_POD="$(kubectl get pods -n kube-system -o wide | grep $(hostname) | awk '{ print $1 }' | grep cilium)"
+
+log "running test: $TEST_NAME"
+
+function finish_test {
+  log "starting finish_test for ${TEST_NAME}"
+  gather_files ${TEST_NAME} k8s-tests
+  gather_k8s_logs "2" ${LOGS_DIR}
+  log "finished running test: $TEST_NAME"
+}
+
+trap finish_test exit
+
+log "importing CNP missing endpointSelector"
+k8s_apply_policy kube-system create "${bad_policy_path}"
+
+log "waiting 2 seconds for the policy to get to cilium-agent"
```

Comment:
`k8s_apply_policy` will wait for the policy to be applied

Reply:
So it does! our tests also told me to not use sleep ;) so that was handy.